mevolib/inference/_FastTree.py

```diff
@@ -1,142 +1,120 @@
 
 import os
 import tempfile
+from io import StringIO
+from pathlib import Path
+from typing import Optional
 
+import Bio.Phylo.BaseTree
 from Bio import Phylo
 
-from mevolib._utils import get_abspath
-from mevolib._py3k import StringIO
-
-
-#-------------------------------------------------------------------------------
-
-SPRT_INFILE_FORMATS = ['fasta', 'phylip']
+SPRT_INFILE_FORMATS = ["fasta", "phylip"]
 
-KEYWORDS = {'default': ['-gtr', '-nt', '-nopr', '-quiet'],
-            # Nucleotide evolution models
-            'GTR+CAT': ['-gtr', '-nt', '-nopr', '-quiet'],
-            'GTR+G':   ['-gtr', '-nt', '-nocat', '-gamma', '-nopr', '-quiet'],
-            'JC+CAT':  ['-nt', '-nopr', '-quiet'],
-            'JC+G':    ['-nt', '-nocat', '-gamma', '-nopr', '-quiet'],
-            # Protein evolution models
-            'JTT+CAT': ['-nopr', '-quiet'],
-            'WAG+CAT': ['-wag', '-nopr', '-quiet']}
+KEYWORDS = {
+    "default": ["-gtr", "-nt", "-nopr", "-quiet"],
+    # Nucleotide evolution models
+    "GTR+CAT": ["-gtr", "-nt", "-nopr", "-quiet"],
+    "GTR+G": ["-gtr", "-nt", "-nocat", "-gamma", "-nopr", "-quiet"],
+    "JC+CAT": ["-nt", "-nopr", "-quiet"],
+    "JC+G": ["-nt", "-nocat", "-gamma", "-nopr", "-quiet"],
+    # Protein evolution models
+    "JTT+CAT": ["-nopr", "-quiet"],
+    "WAG+CAT": ["-wag", "-nopr", "-quiet"],
+}
 
 
-#-------------------------------------------------------------------------------
-
-def gen_args ( args, infile_path, bootstraps ) :
+def gen_args(args: str, infile_path: str, bootstraps: int, log_tmpfile: Optional[str] = None) -> list:
     """
     Return the argument list generated from 'args', the infile path and the
     bootstraps requested.
 
     Arguments :
-        args  ( string )
-            Keyword or arguments to use in the call of FastTree, excluding
+        args: Keyword or arguments to use in the call of FastTree, excluding
             infile and outfile arguments.
-        infile_path  ( string )
-            Input alignment file path.
-        bootstraps  ( int )
-            Number of bootstraps to generate.
+        infile_path: Input alignment file path.
+        bootstraps: Number of bootstraps to generate.
+        log_tmpfile: Path of the directory we want to save the FastTree output data into (Only required while testing for
+            reproducibility purposes or just if the user wants a specific folder to allocate the result of the execution
+            into. Otherwise, a random one will be provided).
 
     Returns :
-        list
-            List of arguments (excluding binary file) to call FastTree.
+        list: List of arguments (excluding binary file) to call FastTree.
     """
-    if ( args in KEYWORDS ) :
+    if args in KEYWORDS:
         argument_list = list(KEYWORDS[args])
-    else : # args not in KEYWORDS
-        argument_list = [arg  for arg in args.split(' ')]
+    else:  # args not in KEYWORDS
```

Comment:
```suggestion
    else:
```
Those types of comments are not useful.